src/menu.rs

```diff
@@ -0,0 +1,52 @@
+use crate::types::{MenuAction, Model, RunningState};
+use std::io;
+use std::time::Duration;
+
+use ratatui::{
+    crossterm::event::{self, Event, KeyCode},
+    widgets::Paragraph,
+    Frame,
+};
+
+pub fn view(_model: &mut Model, frame: &mut Frame) {
+    frame.render_widget(
+        Paragraph::new("Press space to start\nPress esc to Quit".to_string()),
+        frame.area(),
+    );
+}
+
+pub fn update(model: &mut Model, msg: MenuAction) -> Option<MenuAction> {
+    match msg {
+        MenuAction::StartGame => {
+            model.running_state = RunningState::Game;
+        }
+        MenuAction::Quit => {
+            // You can handle cleanup and exit here
+            model.running_state = RunningState::Done;
+        }
+    };
+    None
+}
```

Comment:
Can you remove `-> Option<MenuAction>` here because you only ever return `None`?

Reply:
I think updates type if it were a trait would be `-> Option<T>`. The reasoning is that you can optionally return a `MenuAction` if you need to run two update states in a row. It's not used yet but It's fundamental to the design.